src/modules/ui/directives/copy/copy.js

```diff
@@ -2,54 +2,64 @@
     'use strict';
 
     /**
+     * @param {Base} Base
+     * @param $scope
      * @param {app.utils} utils
      * @param {NotificationManager} notificationManager
-     * @return {*}
+     * @param {JQuery} $element
      */
-    const directive = function (utils, notificationManager) {
-        return {
-            restrict: 'A',
-            priority: 100,
-            scope: false,
-            compile: function ($element, $attrs) {
-                return function ($scope, $element) {
-
-                    const destroy = new tsUtils.Signal();
-                    const stop = $scope.$on('$destroy', () => {
-                        destroy.dispatch();
-                        stop();
-                    });
+    const controller = function (Base, $scope, utils, notificationManager, $element) {
 
-                    const param = utils.parseAngularParam($attrs.wCopy, $scope, destroy);
-                    const clipboard = new Clipboard($element.get(0), {
-                        text: () => param.value,
-                        container: $element.get(0)
-                    });
+        class Copy extends Base {
 
-                    clipboard.on('success', () => {
-                        notificationManager.info({
-                            ns: 'app.ui',
-                            title: { literal: 'copySuccess' }
-                        });
-                    });
+            constructor() {
+                super($scope);
+                const clipboard = new Clipboard($element.get(0), {
+                    text: () => $scope.wCopy,
+                    container: $element.get(0)
```

Comment:
`$element.get(0)` might be placed in a variable `container` and used both here and on line 17.